OpenMcdf.Ole/OlePropertiesContainer.cs

```diff
@@ -127,6 +127,49 @@ public OleProperty CreateProperty(VTPropertyType vtPropertyType, uint propertyId
 
     public void Add(OleProperty property) => properties.Add(property);
 
+    /// <summary>
+    /// Create a new UserDefinedProperty.
+    /// </summary>
+    /// <param name="vtPropertyType">The type of property to create.</param>
+    /// <param name="name">The name of the new property.</param>
+    /// <returns>The new property.</returns>
+    /// <exception cref="InvalidOperationException">If UserDefinedProperties aren't allowed for this container.</exception>
+    /// <exception cref="ArgumentException">If a property with the name <paramref name="name"/> already exists."/></exception>
+    public OleProperty AddUserDefinedProperty(VTPropertyType vtPropertyType, string name)
+    {
+        if (this.ContainerType != ContainerType.UserDefinedProperties)
+            throw new InvalidOperationException();
+
+        // As per https://learn.microsoft.com/en-us/openspecs/windows_protocols/MS-OLEPS/4177a4bc-5547-49fe-a4d9-4767350fd9cf
+        // the property names have to be unique, and are case insensitive.
+        if (PropertyNames.Any(property => property.Value.Equals(name, StringComparison.InvariantCultureIgnoreCase)))
```

Comment:
@Numpsy Thanks for the review! I enabled nullability warnings for 3.0. The warnings here look legitimate, since PropertyNames isn't initialized in the constructor. I did the least amount of work to satisfy the compiler - i.e. I added the missing attributes rather than fixed the initialization. If you want to fix up the nullability attributes/initialization accordingly in a later commit, then that would be great. You'd know much more about the expectations about whether something should/shouldn't be allowed to be null - I've merely enabling the enforcement :)